### ISSUE TITLE
ideas to improve the project

### ISSUE THREAD

**Issue by CherkasovaViktoria** [CherkasovaViktoria/churn-rate#2]
1.Добавить  criterion = 'gini' и min_samples_leaf в модель  DecisionTreeClassifier на всех этапах исследования
2. Обучить модели при помощи кросс-валидации.
3. Распределение по годам использования банка практически ровное, если сейчас заменить 909 пропусков медианой, то мы увеличим в 2 раза значение для 5 лет. Это кажется некорректным. Можно сделать заполнение пропусков случайными числами, тогда мы снизим влияние на наше распределение.
Так как значений в этом признаке ограниченное количество, можно сделать его категориальным признаком. Пропуски можно считать как за отдельную категорию (заполнить значением -1). Затем заменить тип данных Tenure на object и применить технику OHE.
4. Исследовать выбросы в количественных признаках и по возможности привести их к нормальному распределению для улучшения предсказаний модели LogisticRegression при помощи масштабирования, удаления ненужных записей.
5. При разбиении выборок при помощи train_test_split использовать параметр shuffle для более равномерного распределения баланса классов.
6. Использовать GridSearchCV для подбора лучших гиперпараметров.
7. Проверить корреляцию признаков при необходимости удалить ненужные признаки.
